cylibs/trust/roles/puller.lua

```diff
@@ -186,50 +169,66 @@ end
 
 function Puller:get_next_target()
     if state.AutoPullMode.value == 'Party' then
+        -- Get all targets the party is tracking
         local party_targets = self:get_party():get_targets(function(t)
+            -- Filter to distance 25, and engaged
             return t:get_distance():sqrt() < 25 and t:get_mob().status == 1
         end):sort(function(t1, t2)
+            -- Sort by distance
             return t1:get_distance() < t2:get_distance()
         end)
+        -- If the party is tracking any targets
         if party_targets:length() > 0 then
+            -- Get a list of all targets the party is targeting
             local party_target_indices = self:get_party():get_party_members(false):map(function(p)
                 return p:get_target_index()
             end):compact_map()
-            local next_target = party_targets:firstWhere(function(target)
+            -- Select next target, prefer targets not targeted by an ally
+            local next_targets = party_targets:filter(function(target)
                 return target and not party_target_indices:contains(target:get_mob().index)
-            end) or party_targets[1]
+            end)
+            local next_target = next_targets[math.random(math.min(math.max(1, next_targets:length()), 6))] or party_targets[math.random(math.min(math.max(1, party_targets:length()), 6))]
             local monster = Monster.new(next_target:get_id())
             return monster
         end
     else
-        -- First, check mobs that are aggroed and unclaimed or aggroed and claimed by a party member
-        local nearby_mobs = ffxi_util.find_closest_mobs(L{}, L{}, L{}, 10):filter(function(mob)
-            if res.statuses[mob.status].en == 'Engaged' then
-                if mob.claim_id == 0 then
-                    logger.notice(self.__class, 'get_next_target', 'engaged', 'unclaimed')
-                    return true
-                else
-                    if self:get_party():get_party_member(mob.claim_id) then
-                        logger.notice(self.__class, 'get_next_target', 'engaged', 'claimed')
+        -- Ensure that we are either in all mode, or that target list is populated
+        if self:get_target_names():length() > 0 or state.AutoPullMode.value == 'All' then
+            -- First, check mobs that are aggroed and unclaimed
+            local nearby_mobs = ffxi_util.find_closest_mobs(L{}, L{}, L{}, 10):filter(function(mob)
```

Comment:
What happens in the case where I pull a mob and aggro a mob (e.g. in Sheol C) and reload Trust. Would it go for the unclaimed mob rather than the one I pulled?

Reply:
Yes, there's no way to know the state from having the addon loaded previously. I suspect that's the only way to get a double pull is reloading it and puller will probably attempt to pull a new unclaimed mob. Granted, if you switch targets, your old one will go unclaimed so someone will pick it up next

Reply:
The reason I ask is because currently it will pick the closest unclaimed aggroed mob or party claimed mob, but with this code it will pick a random one which might end up in a double pull. But I suppose if we set max num mobs to 1 then it should still just pick the closest mob which is probably the one you are currently fighting?

Reply:
Yes, the only way you could get a double pull in party mode is if you reload while you have unclaimed aggroed mobs

The only way to get a double pull in auto/all mode is to reload while there are unclaimed mobs nearby.

If we really want to go down that rabbit hole, we could add a check for if you're currently engaged to a target, and have it ignore pulling if so